compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4766,4 +4807,141 @@ private boolean isConst(BLangExpression expression) {
     private Name getCurrentCompUnit(BLangNode node) {
         return names.fromString(node.pos.getSource().getCompilationUnitName());
     }
+
+    private BType getRepresentativeBroadType(List<BLangExpression> exprs) {
+        LinkedHashSet<BType> narrowTypes = new LinkedHashSet<>();
+        LinkedHashSet<BType> broadTypesSet = new LinkedHashSet<>();
+        BType[] inferredTypes = checkExprList(exprs, env);
+        for (BType type : inferredTypes) {
+            if (type.tag == TypeTags.SEMANTIC_ERROR) {
+                return type;
+            }
+
+            if (narrowTypes.stream().noneMatch(nType -> types.isSameType(type, nType))) {
+                narrowTypes.add(type);
+            }
+        }
+        BType broadType;
+        for (BType t1 : narrowTypes) {
+            broadType = t1;
+            for (BType t2 : narrowTypes) {
```

Comment:
Should the second loop start from the t1+1 item to reduce duplicate ```isAssignable``` calls?

Reply:
I've updated the `getRepresentativeBroadType()` logic a bit with ddbbe63. Those changes should address these concerns.